nautobot/core/models/fields.py

```diff
@@ -415,3 +417,48 @@ def formfield(self, form_class=fields.DynamicModelMultipleChoiceField, **kwargs)
         kwargs.setdefault("required", False)
         kwargs.setdefault("query_params", {"content_types": self.model._meta.label_lower})
         return super().formfield(form_class=form_class, **kwargs)
+
+
+class PositiveRangeNumberTextField(models.TextField):
+    default_error_messages = {
+        "invalid": "Invalid value. Specify a value using positive integers in a range format (i.e. '10-20').",
```

Comment:
Zero is valid for this field in general, right?

```suggestion
        "invalid": "Invalid value. Specify a value using non-negative integers in a range format (i.e. '10-20').",
```